### PR TITLE
Applies fixes for Doc Parser.

### PR DESCRIPTION
**Pull request by BinarySoftware**
### Pull Request Description
This PR bumps scala parser version to include latest commit to parser from engine repository, which fixes documentation tags, adds new tags according to specs as well as adds some of unavailable character, so that every doc in stdlib parses.

### Important Notes
This PR wont solve issue with indent in first line, it will be fixed after updating AST.

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Rust](https://github.com/enso-org/enso/blob/main/docs/style-guide/rust.md) style guide.
- [x] All code has automatic tests where possible.
- [x] All code has been profiled where possible.
- [x] All code has been manually tested in the IDE.
- [x] All code has been manually tested in the "debug/interface" scene.
- [x] All code has been manually tested by the PR owner against our [test scenarios](https://docs.google.com/spreadsheets/d/1RatJDM_f9_3bvYhl3Bpq2d8SyKgtVdrV1RkGxPU17c8/edit?ts=5faa7049#gid=0).
- [x] All code has been manually tested by at least one reviewer against our [test scenarios](https://docs.google.com/spreadsheets/d/1RatJDM_f9_3bvYhl3Bpq2d8SyKgtVdrV1RkGxPU17c8/edit?ts=5faa7049#gid=0).

### CODE REVIEW

File: src/rust/ide/view/src/documentation.rs [hunk 1]
```diff
@@ -96,7 +96,7 @@ impl Model {
         let bg_color = styles.get_color(ensogl_theme::graph_editor::visualization::background);
         let bg_color = color::Rgba::from(bg_color);
         let bg_hex   = format!("rgba({},{},{},{})",
-            bg_color.red*255.0,bg_color.green*255.0,bg_color.blue*255.0,bg_color.alpha);
+                               bg_color.red*255.0,bg_color.green*255.0,bg_color.blue*255.0,bg_color.alpha);
```
Comment (wdanilo) [inline src/rust/ide/view/src/documentation.rs]: strange indentation, we use 4 spaces indentation, here is way more

**Review by wdanilo — APPROVED** [review]

**Review by farmaazon — COMMENTED** [review]
I see issues in some docs:
1. A dot in a new line.
image 
2. One word became an example:
image

Are they a bugs in a docstrings, or in doc parser?

**Review by wdanilo — CHANGES_REQUESTED** [review]
image

With this (lack of) description I completely do not understand. what this PR really does.

**Comment by BinarySoftware** [comment]
> I see issues in some docs:
> 
> 1. A dot in a new line.
>    image
> 2. One word became an example:
>    image
> 
> Are they a bugs in a docstrings, or in doc parser?

Hey Adam, those are bugs that i simply cant fix right now, as easily as i've fixed the tags or unavailable chars.

**Comment by BinarySoftware** [comment]
> image
> 
> With this (lack of) description I completely do not understand. what this PR really does.

Updated.

**Review by wdanilo — CHANGES_REQUESTED** [review]
Maciek, I asked you on Discord to create missing issues and link them here as a reply to Adam, it is not done yet, so I'm requesting changes.

**Comment by BinarySoftware** [comment]
> Maciek, I asked you on Discord to create missing issues and link them here as a reply to Adam, it is not done yet, so I'm requesting changes.

https://github.com/enso-org/enso/issues/1441

File: src/rust/ide/view/src/documentation.rs [hunk 2]
```diff
@@ -141,10 +141,17 @@ impl Model {
...
+            let parser        = parser::DocParser::new()?;
+            let mut processed = string.to_string();
+            let lines = processed.lines().collect::<Vec<&str>>();
+            if lines.len() > 1 {
+                let mut indent = 0;
+                for char in lines[1].chars() {
+                    if char != ' ' { break; }
+                    indent  += 1;
+                }
+                processed = format!{"{}{}",String::from(" ").repeat(indent),processed};
```
Comment (wdanilo) [inline src/rust/ide/view/src/documentation.rs]: this code is magical, it has not documentation and there is no explanation what it really does. It is inside a function `gen_html_from`, but it definitely is not generating HTML, so it needs proper documentation / refactoring to other, well named function

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,7 +2,9 @@
 This is a minor update exposing more control over the anonymous data collection. You are now allowed 
 to run the application with `--no-errors-data-collection`, and `--no-data-collection`. The options 
 disable the collection of errors data (keeping the collection of other events), and disables the 
-collection of all data, respectively. To learn more what data is collected in Enso alpha releases, 
+collection of all data, respectively. This release also includes updated documentation parser,
+with hotfixes for things such as misinterpreted tags or unknown characters that are commonly used.
+To learn more what data is collected in Enso alpha releases, 
 read the notes of the `Enso 2.0.0-alpha.1` release.
```
Comment (wdanilo) [inline CHANGELOG.md]: Put that pls in the `bug` fixes section (according to the CHANGELOG_TEMPLATE.md template - with link to your PR).

**Review by wdanilo — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,7 +2,9 @@
 This is a minor update exposing more control over the anonymous data collection. You are now allowed 
 to run the application with `--no-errors-data-collection`, and `--no-data-collection`. The options 
 disable the collection of errors data (keeping the collection of other events), and disables the 
-collection of all data, respectively. To learn more what data is collected in Enso alpha releases, 
+collection of all data, respectively.
+  
+To learn more what data is collected in Enso alpha releases, 
 read the notes of the `Enso 2.0.0-alpha.1` release.
```
Comment (farmaazon) [inline CHANGELOG.md]: This line is strangely wrapped. Please wrap it near the spot where other lines are wrapped.

File: CHANGELOG.md [hunk 5]
```diff
@@ -16,6 +18,8 @@ read the notes of the `Enso 2.0.0-alpha.1` release.
   now on all supported platforms.
 - [You can now see data frames in the table visualisation][1181]. Big tables get truncated to 2000 
   entries.
+- [Updated documentation parser][1098] with hotfixes for things such as misinterpreted tags or 
+  unknown characters that are commonly used.
```
Comment (farmaazon) [inline CHANGELOG.md]: I'm not sure if user will know what is "documentation parser". Let's name it "Fixing documentation layout in Searcher" as this is currently the visible change.

File: src/rust/ide/view/src/documentation.rs [hunk 6]
```diff
@@ -154,6 +152,22 @@ impl Model {
         }
     }
 
+    /// Fixes indentation in the first line of documentation by looking at next line, as documentation
+    /// doesn't know what was its indentation in the Enso source file.
```
Comment (farmaazon) [inline src/rust/ide/view/src/documentation.rs]: Please extend this description. What does "looking at next line" mean? Do we adjust the first line to the second? What if there is only one line? What if documentation author meant the second line to be indented?

If we really need to put a workaround here (not in parser), then I'd rather put three spaces to the first line (an equivalent of `## `. Or maybe shall we put two spaces for `##` instead? I'm not sure, I don't know how parser works).

File: src/rust/ide/view/src/documentation.rs [hunk 7]
```diff
@@ -154,6 +152,22 @@ impl Model {
...
+    fn fix_indent_in_first_line(string: &str) -> String {
```
Comment (farmaazon) [inline src/rust/ide/view/src/documentation.rs]: Let it take `impl Into<String>` instead.

File: src/rust/ide/view/src/documentation.rs [hunk 8]
```diff
@@ -154,6 +152,22 @@ impl Model {
...
+        let mut processed = string.to_string();
+        let lines = processed.lines().collect::<Vec<&str>>();
```
Comment (farmaazon) [inline src/rust/ide/view/src/documentation.rs]: You can do that without allocating a vector. The most smart way I made out is:
```rust
let lines = processed.lines().take(2).collect<SmallVec<[&str;2]>>();
```
It of course borrow `processed`, but it does not have to borrow still when we use `processed` again.

File: src/rust/ide/view/src/documentation.rs [hunk 9]
```diff
@@ -154,6 +152,22 @@ impl Model {
...
+        if lines.len() > 1 {
+            let mut indent = 0;
+            for char in lines[1].chars() {
+                if char != ' ' { break; }
+                indent += 1;
+            }
```
Comment (farmaazon) [inline src/rust/ide/view/src/documentation.rs]: It can be replaced with one-liner:
```rust
let ident = lines[1].chars().take_while(|c| c == ' ').count();
```

**Review by farmaazon — CHANGES_REQUESTED** [review]
I see still one not-working case in the workaround provided by this PR:
```
## A function foo doing foo like this:
       foo
```
Here the doc writer meant the second line to be intended, but we align the first line to the second.

I could accept this PR, as it is not a regression, but first confirm with @wdanilo that fixing that properly in the parser is impossible or too costly.

File: src/rust/ide/view/src/documentation.rs (hunk 6 above)
Comment (BinarySoftware) [reply src/rust/ide/view/src/documentation.rs]: no, putting fixed amount of spaces wont fix the issue.
Take a look at some enso code:
```py
type foo
    type bar
        ## docs here
           docs there
        type baz
```
We need to add the indentation of whole block plus the indent of 2 hashes - that is in this case 10 additional spaces, otherwise doc parser will get code like this:
```
 docs here
          docs there
```
and will create a synopsis with raw text "docs here" and code block of  "docs there".

As we talked on dm, if there is only one line, there is no problem, it just wont do anything, and will work. If the next line is empty, there is still no problem, as what is in the next box will just become new paragraph, just as intended, in "body" section. 

The only corner case is when the user specifically intends to make second line a code block, but that must be a really rare case, as i havent seen such in any doc in stdlib.

To sum up, this thing is really not a trivial fix as the doc code we get doesnt know anything about its origin, what was its indentation etc, hence the need to write such a hack. To be further discussed with @wdanilo but I dont see any other reasonable solution currently.

Comment (BinarySoftware) [reply src/rust/ide/view/src/documentation.rs]: After talk I've removed hack method, this PR wont fix indent in first line.

File: CHANGELOG.md [hunk 10]
```diff
@@ -16,6 +16,8 @@ read the notes of the `Enso 2.0.0-alpha.1` release.
   now on all supported platforms.
 - [You can now see data frames in the table visualisation][1181]. Big tables get truncated to 2000 
   entries.
+- [Fixing documentation layout in Searcher][1098]. Hotfixes for things such as misinterpreted tags 
+  and unknown characters that are commonly used.
```
Comment (farmaazon) [inline CHANGELOG.md]:
```suggestion
- [Fixing documentation layout in Searcher][1098]. Includes misinterpreted tags and unknown characters that are commonly used.
```

**Review by farmaazon — APPROVED** [review]
Please fix a changelog entry before merge. I would rather not mention that our fixes are "hot".

File: CHANGELOG.md [hunk 11]
```diff
@@ -16,6 +16,7 @@ read the notes of the `Enso 2.0.0-alpha.1` release.
   now on all supported platforms.
 - [You can now see data frames in the table visualisation][1181]. Big tables get truncated to 2000 
   entries.
+- [Fixing documentation layout in Searcher][1098]. Includes misinterpreted tags and unknown characters that are commonly used.
```
Comment (wdanilo) [inline CHANGELOG.md]: the changelog entries should be understandable to the end-users. Not all are perfect now, but this description "fixing documentation layout in Searcher" is not very understandable. Also, "fixing" means that it is being done, not done already, so its not a good form for a changelog. Can we describe here what was fixed and what kind of docs will be displayed better now ?

File: build/run.js [hunk 12]
```diff
@@ -171,7 +171,7 @@ commands.build.rust = async function(argv) {
 
         console.log('Checking the resulting WASM size.')
         let stats = fss.statSync(paths.dist.wasm.mainOptGz)
-        let limit = 4.29
+        let limit = 4.37
```
Comment (wdanilo) [inline build/run.js]: This is blocker to me. This is HUGE change. Where it comes from??? Remember, that parser is NOT included in this file size, so what Rust code changes were here that caused this increase?

**Review by wdanilo — CHANGES_REQUESTED** [review]

**Review by wdanilo — APPROVED** [review]
